date_range/models/date_range_type.py

```diff
@@ -22,6 +22,9 @@ def _default_company(self):
     company_id = fields.Many2one(
         comodel_name='res.company', string='Company', index=1,
         default=_default_company)
+    parent_type_id = fields.Many2one(
```

Comment:
We should declare a constrains to avoid cycle into the parent_type_id relation...

Reply:
@ntsirintanis The implementation is not complete. It will not detect Type A `parent_of` Type B `parent_of` Type C `parent_of` Type A

Reply:
@lmignon thanks so much for this tip! We are really really close to finishing this.